odoo/modules/registry.py

```diff
@@ -231,6 +231,9 @@ def setup_models(self, cr):
         lazy_property.reset_all(self)
         env = odoo.api.Environment(cr, SUPERUSER_ID, {})
 
+        if env.all.tocompute:
```

Comment:
I wonder if we shouldn't flush here as a one line fix instead

Reply:
You don't know whether the registry is in a correct state at this point.
Blindly flushing is unsafe and would simply hide a programming error.